src/Stack/ConfigCmd.hs

```diff
@@ -0,0 +1,71 @@
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+
+-- | Make changes to the stack yaml file
+
+module Stack.ConfigCmd
+       (ConfigCmdSet(..)
+       ,cfgCmdSet
+       ,cfgCmdSetName
+       ,cfgCmdName) where
+
+import           Control.Monad.Catch (MonadMask, throwM, MonadThrow)
+import           Control.Monad.IO.Class
+import           Control.Monad.Logger
+import           Control.Monad.Reader (MonadReader, asks)
+import           Control.Monad.Trans.Control (MonadBaseControl)
+import qualified Data.ByteString.Builder         as B
+import qualified Data.ByteString.Lazy            as L
+import qualified Data.HashMap.Strict as HMap
+import qualified Data.Yaml as Yaml
+import           Network.HTTP.Client.Conduit (HasHttpManager)
+import           Path
+import           Stack.BuildPlan
+import           Stack.Init
+import           Stack.Types
+
+data ConfigCmdSet = ConfigCmdSetResolver AbstractResolver
+
+cfgCmdSet :: ( MonadIO m
+             , MonadBaseControl IO m
+             , MonadMask m
+             , MonadReader env m
+             , HasConfig env
+             , HasBuildConfig env
+             , HasHttpManager env
+             , HasGHCVariant env
+             , MonadThrow m
+             , MonadLogger m)
+             => ConfigCmdSet -> m ()
+cfgCmdSet (ConfigCmdSetResolver newResolver) = do
+    stackYaml <- bcStackYaml <$> asks getBuildConfig
+    let stackYamlFp =
+            toFilePath stackYaml
+    -- We don't need to worry about checking for a valid yaml here
+    (projectYamlConfig :: Yaml.Object) <-
+        liftIO (Yaml.decodeFileEither stackYamlFp) >>=
+        either throwM return
+    newResolverText <- resolverName <$> makeConcreteResolver newResolver
+    -- We checking here that the snapshot actually exists
+    snap <- parseSnapName newResolverText
+    _ <- loadMiniBuildPlan snap
+
+    let projectYamlConfig' =
+            HMap.insert
+                "resolver"
+                (Yaml.String newResolverText)
+                projectYamlConfig
+    liftIO
```

Comment:
More efficient: just use `Yaml.encodeFile`. Definitely not necessary to convert from a strict `ByteString` to a lazy `ByteString` via a `Builder`.